src/MiddlewareRegistry.ts

```diff
@@ -81,4 +82,24 @@ export class MiddlewareRegistry<R extends MiddlewareRequest> {
         return (this.request as NextApiRequest).url;
     
     }
+
+    /**
+     * Since different middleware may be added for the same route but under different HTTP verbs, a composite key
+     * is required for the middleware mapping.
+     * @param route Route passed when added to the registry
+     * @param methods Methods, if any, to distinguish the entry by
+     * @private
+     */
+    private serializeToRegistryKey(route: string, methods?: MiddlewareConfig['methods']): string {
+        return JSON.stringify({ route, methods });
+    }
+
+    /**
+     * Given a serialized key from the middleware mapping, return the route
+     * @param serializedKey The serialized key to parse
+     * @private
+     */
+    private deserializeKeyToPathAndMethods(serializedKey: string): string {
```

Comment:
`deserializeKeyToRoute` maybe?

Reply:
Ah yes, remnant of an earlier version. Will do, thanks.